src/bin/cargo/cli.rs

```diff
@@ -691,6 +693,20 @@ See '<cyan,bold>cargo help</> <cyan><<command>></>' for more information on a sp
                 }))
             }).collect()
         })))
+        .add(clap_complete::engine::SubcommandCandidates::new(move || {
+            alias_map.iter()
+                .map(|(alias, cmd_info)| {
+                    let help_text = if let super::CommandInfo::Alias { target } = cmd_info {
+                        let cmd_str = target.iter().map(String::as_str).collect::<Vec<_>>().join(" ");
+                        format!("alias for {}", cmd_str)
+                    } else {
+                        "alias (from config)".to_string()
+                    };
```

Comment:
I take it this branch is a "just in case" a new variant is added relevant for aliases and we don't want to break the user?  Otherwise, it looks like this could be an `unreachable!`.

One option is to use a `match` and make each of the existing variants `unreachable`.  If a new variant is added, it will then be a compiler error to let us know to update this by either reporting a candidate or by adding `unreachable!`